express-app/controllers/user.js

```diff
@@ -109,4 +127,21 @@ module.exports = {
       console.log(error);
     });
   },
+
+  /**
+  * This function reset the password from a user that was
+  * actually logged.
+  *
+  * @function
+  * @param {object} user - actual user who will be edited.
+  * @param {object} data - user updated
+  */
+  resetPassword: function (user, data) {
+    let hash = bcrypt.hashSync(data.password, bcrypt.genSaltSync(10), null);
+    User.findOneAndUpdate({ _id: user._id }, { password: hash }).then((editedUser) => {
```

Comment:
Have you tested this at least manually, I'm not sure if when the password is reset, it would be hashed 2 times or not? This I think because you are hashing the password weach time the user is saved right?